x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/reducer.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// eslint-disable-next-line import/no-nodejs-modules
+import { parse } from 'querystring';
+import { matchPath } from 'react-router-dom';
+import { ImmutableReducer } from '../../../../common/store';
+import { AppLocation, Immutable } from '../../../../../common/endpoint/types';
+import { UserChangedUrl } from '../../../../common/store/routing/action';
+import { extractListPaginationParams } from '../../../common/routing';
+import {
+  MANAGEMENT_ROUTING_TRUSTED_APPS_PATH,
+  MANAGEMENT_DEFAULT_PAGE,
+  MANAGEMENT_DEFAULT_PAGE_SIZE,
+} from '../../../common/constants';
+import { TrustedAppsListDataBindingChanged, TrustedAppsPageAction } from './action';
+import { TrustedAppsListPageState } from '../state/trusted_apps_list_page_state';
+
+type StateReducer = ImmutableReducer<TrustedAppsListPageState, TrustedAppsPageAction>;
```

Comment:
Ahh. Ok. This answers my earlier question. I suggest you use `AppAction` here instead of `TrustedAppsPageAction`

Reply:
I've made a change but I would like us to have a discussion about this. Current store strategy creates a very tight coupling between all sub-stores, and I can't say I see a very good reason for that. I do see why there would be a need to react on routing action that is not relevant for your app (like resetting everything or making some flag that indicates if page is active), but other than that I see not much benefit of middleware and reducer being invoked for every action that happened out there. It's first of all some performance impact (but most likely negligible) but secondly it is coupling all stores together. Consider moving to ingest - it's gonna be a tough job to detach our code from all other stores. It creates quite complex dependency tree.